src/step_db.cpp

```diff
@@ -351,7 +350,7 @@ bool fields_chunk::step_db(field_type ft) {
         realnum f_m_mult = ft == D_stuff ? 2 : (1 - 2 * cmp) * m;
         realnum dt2 = dt * 0.5;
 
-        for (int iz = (ft == D_stuff); iz < nz + (ft == D_stuff); ++iz) {
+        for (int iz = (ft == D_stuff); iz < nz; ++iz) {
```

Comment:
This is a little surprising to me.  I thought that `iz == nz` was owned for the `Dp` component?   This means that `Dp` has only `nz-1` owned points, whereas `Hr` has `nz` owned points?

Would be worth printing out `gv.little_owned_corner(Dp)` and `gv.big_owned_corner(Dp)` and checking how those values relate to these (after you divide by 2) in the `z` component.

Reply:
Adding some `printf` statements to `step_db.cpp` for outputting the properties of the chunk's `grid_volume` `gv` for the PML and non-PML regions used during the `Dp` and `Hr` field updates:
```
--- a/src/step_db.cpp
+++ b/src/step_db.cpp
@@ -350,6 +350,22 @@ bool fields_chunk::step_db(field_type ft) {
         realnum f_m_mult = ft == D_stuff ? 2 : (1 - 2 * cmp) * m;
         realnum dt2 = dt * 0.5;
 
+        if (sig || sigu)
+          printf("PML region\n");
+        else
+          printf("non PML region\n");
+        printf("vol:, (%0.2f, %0.2f), (%0.2f, %0.2f)\n",
+               gv.surroundings().in_direction_min(R), gv.surroundings().in_direction_min(Z),
+               gv.surroundings().in_direction_max(R), gv.surroundings().in_direction_max(Z));
+        printf("grid_vol:, Dp, (%d, %d), (%d, %d), %d, %d\n",
+               gv.little_owned_corner(Dp).r(), gv.little_owned_corner(Dp).z(),
+               gv.big_owned_corner(Dp).r(), gv.big_owned_corner(Dp).z(),
+               gv.big_owned_corner(Dp).z() - gv.little_owned_corner(Dp).z() + 1, nz);
+        printf("grid_vol:, Hr, (%d, %d), (%d, %d), %d, %d\n",
+               gv.little_owned_corner(Hr).r(), gv.little_owned_corner(Hr).z(),
+               gv.big_owned_corner(Hr).r(), gv.big_owned_corner(Hr).z(),
+               gv.big_owned_corner(Hr).z() - gv.little_owned_corner(Hr).z() + 1, nz);
+
```
Then running the unit test `python3 test_ldos.py TestLDOS.test_ldos_ext_eff` (`Er` source at $r = 0$ for $m = -1$) shows the following output:

```
Working in Cylindrical dimensions.
Computational cell is 6.52 x 0 x 1.72 with resolution 25
     block, center = (0,0,-0.75)
          size (1e+20,1e+20,0.208333)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (5.76,5.76,5.76)
time for set_epsilon = 0.0203028 s
-----------
Meep: using complex fields.
PML region
vol:, (0.00, 0.32), (5.96, 0.88)
gv:, Dp, (0, 18), (298, 44), 27, 14
gv:, Hr, (0, 17), (298, 43), 27, 14
non PML region
vol:, (0.00, -0.84), (5.96, 0.32)
gv:, Dp, (0, -40), (298, 16), 57, 29
gv:, Hr, (0, -41), (298, 15), 57, 29
```

![led_plot2D](https://github.com/NanoComp/meep/assets/7152530/579f0ff3-0e85-4119-9138-b81382c03cfd)

The `Dp` and `Hr` field arrays each contains 14 and 29 grid points in the $z$ direction within the PML and non-PML regions, respectively. The difference in the $z$ coordinate of the `big_owned_corner` and `little_owned_corner` is 27 and 57, respectively.